site/en/posts/android-package-signer.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Introducing Android Package Signer
+metadesc: Learn about signing Android packages client side in your browser with this new library.
+tags:
+  - technical
+authors:
+  - nohe
+date: 2022-01-26
+---
+
+The Android Package Signer library is a new open-source JavaScript library that allows web developers to both generate signing keys and use signing keys to sign Android packages, all entirely in browser. It's designed to allow anyone who runs app building as a service to do so without requiring users to upload their credentials, an important part of [keeping signing keys secure](https://developer.android.com/studio/publish/app-signing#secure_key). As a bonus, this package also removes a dependency on Java for developers looking to sign their pre-built unsigned Android packages.
+
+## Android Package Signer and Bubblewrap
+
+[Bubblewrap](https://github.com/GoogleChromeLabs/bubblewrap) is a project created to help web developers package their Progressive Web Apps for distribution on Google Play and has been used to be the foundation of app bundling services. Often, though, this meant users were handing over the Android signing keys and password when creating a new project without knowing if they were being stored or potentially leaked. Having a leaked key could allow a malicious party to impersonate a developer and release harmful package updates that would otherwise appear to be legitimate.
+
+## Installation and Usage
+
+The [Android Package Signer repository](https://github.com/chromeos/android-package-sign-js) is the best place to go to keep up-to-date with the package.
+
+To add Android Package Signer to your project, first install it from NPM:
+
+````bash {title="bash" .code-figure}
+npm i @chromeos/android-package-signer
+``` From there, require it in your project and initialize it:
```

Comment:
This is what I was afraid of with the in-line suggestion. I believe this needs to be edited by-hand to move this sentence to the next line